Index: xpra/codecs/jpeg/decoder.pyx
===================================================================
--- a/xpra/codecs/jpeg/decoder.pyx	(revision 21967)
+++ b/xpra/codecs/jpeg/decoder.pyx	(working copy)
@@ -31,7 +31,7 @@
     TJSAMP  TJSAMP_420
     TJSAMP  TJSAMP_GRAY
     TJSAMP  TJSAMP_440
-    TJSAMP  TJSAMP_411
+    #TJSAMP  TJSAMP_411
 
     TJPF    TJPF_RGB
     TJPF    TJPF_BGR
@@ -44,7 +44,7 @@
     TJPF    TJPF_BGRA
     TJPF    TJPF_ABGR
     TJPF    TJPF_ARGB
-    TJPF    TJPF_CMYK
+    #TJPF    TJPF_CMYK
 
     TJCS    TJCS_RGB
     TJCS    TJCS_YCbCr
@@ -84,15 +84,15 @@
     TJSAMP_420  : "420",
     TJSAMP_GRAY : "GRAY",
     TJSAMP_440  : "440",
-    TJSAMP_411  : "411",
+    #TJSAMP_411  : "411",
     }
 
 TJCS_STR = {
-    TJCS_RGB    : "RGB",
+    #TJCS_RGB    : "RGB",
     TJCS_YCbCr  : "YCbCr",
     TJCS_GRAY   : "GRAY",
-    TJCS_CMYK   : "CMYK",
-    TJCS_YCCK   : "YCCK",
+    #TJCS_CMYK   : "CMYK",
+    #TJCS_YCCK   : "YCCK",
     }
 
 TJPF_STR = {
@@ -107,7 +107,7 @@
     TJPF_BGRA   : "BGRA",
     TJPF_ABGR   : "ABGR",
     TJPF_ARGB   : "ARGB",
-    TJPF_CMYK   : "CMYK",
+    #TJPF_CMYK   : "CMYK",
     }
 TJPF_VAL = reverse_dict(TJPF_STR)
 
Index: xpra/codecs/jpeg/encoder.pyx
===================================================================
--- a/xpra/codecs/jpeg/encoder.pyx	(revision 21967)
+++ b/xpra/codecs/jpeg/encoder.pyx	(working copy)
@@ -32,7 +32,7 @@
     TJSAMP  TJSAMP_420
     TJSAMP  TJSAMP_GRAY
     TJSAMP  TJSAMP_440
-    TJSAMP  TJSAMP_411
+    #TJSAMP  TJSAMP_411
 
     TJPF    TJPF_RGB
     TJPF    TJPF_BGR
@@ -45,7 +45,7 @@
     TJPF    TJPF_BGRA
     TJPF    TJPF_ABGR
     TJPF    TJPF_ARGB
-    TJPF    TJPF_CMYK
+    #TJPF    TJPF_CMYK
 
     int TJFLAG_BOTTOMUP
     int TJFLAG_FASTUPSAMPLE
@@ -74,7 +74,7 @@
     "BGRA"  : TJPF_BGRA,
     "ABGR"  : TJPF_ABGR,
     "ARGB"  : TJPF_ARGB,
-    "CMYK"  : TJPF_CMYK,
+    #"CMYK"  : TJPF_CMYK,
     }
 TJSAMP_STR = {
     TJSAMP_444  : "444",
@@ -82,7 +82,7 @@
     TJSAMP_420  : "420",
     TJSAMP_GRAY : "GRAY",
     TJSAMP_440  : "440",
-    TJSAMP_411  : "411",
+    #TJSAMP_411  : "411",
     }
 
 
