--- a/configure.ac	Fri Sep 16 16:15:34 2016
+++ b/configure.ac	Sun Dec 18 14:09:32 2016
@@ -1605,10 +1605,10 @@
 dnl ** Check for kqueue (GIO) **
 dnl ****************************
 kqueue_support=no
-AC_CHECK_HEADERS([sys/event.h],
-[
-	AC_CHECK_FUNCS(kqueue kevent, [kqueue_support=yes])
-])
+#AC_CHECK_HEADERS([sys/event.h],
+#[
+#	AC_CHECK_FUNCS(kqueue kevent, [kqueue_support=yes])
+#])
 
 AM_CONDITIONAL(HAVE_KQUEUE, [test "$kqueue_support" = "yes"])
 
