--- gst-plugins-good-1.6.4.orig/sys/osxaudio/gstosxcoreaudio.c	2015-09-25 15:19:29.000000000 -0700
+++ gst-plugins-good-1.6.4/sys/osxaudio/gstosxcoreaudio.c	2016-05-04 14:32:26.000000000 -0700
@@ -35,6 +35,12 @@
 #include "gstosxcoreaudiohal.c"
 #endif
 
+#ifndef kLinearPCMFormatFlagsSampleFractionShift
+#define kLinearPCMFormatFlagsSampleFractionShift 7
+#endif
+#ifndef kLinearPCMFormatFlagsSampleFractionMask
+#define kLinearPCMFormatFlagsSampleFractionMask (0x3F << kLinearPCMFormatFlagsSampleFractionShift)
+#endif
 
 static void
 gst_core_audio_class_init (GstCoreAudioClass * klass)
