--- a/src/gtkosxapplication_quartz.c	2017-02-23 12:36:11.000000000 +0700
+++ b/src/gtkosxapplication_quartz.c	2017-02-23 12:36:23.000000000 +0700
@@ -294,7 +294,7 @@
   [menuitem release];
   [app_menu addItem: [NSMenuItem separatorItem]];
 
-  menuitem = [[NSMenuItem alloc] initWithTitle: [NSString stringWithFormat: NSLocalizedStringFromTable (@"Quit %@", @"GtkosxApplication", @"Quit menu item title"), appname]
+  menuitem = [[NSMenuItem alloc] initWithTitle: [NSString stringWithFormat: NSLocalizedStringFromTable (@"Disconnect %@", @"GtkosxApplication", @"Quit menu item title"), appname]
 	      action: @selector (terminate:) keyEquivalent: @"q"];
 /*
  * Accounts for the added application name to the Quit menu item in a
