--- bundler/bundler.py.orig	2014-11-04 13:47:09.000000000 -0800
+++ bundler/bundler.py	2014-11-04 13:47:52.000000000 -0800
@@ -73,7 +73,7 @@
         module_version = utils.evaluate_pkgconfig_variables("${pkg:pango:pango_module_version}")
         modulespath = self.project.get_bundle_path("Contents/Resources/lib/pango/" +
                                                    module_version +
-                                                   "modules/")
+                                                   "/modules/")
 
         from distutils.version import StrictVersion as V
         import tempfile
