diff -urN lz4-0.7.0/setup.py lz4-0.7.0.new/setup.py
--- lz4-0.7.0/setup.py	2014-07-07 20:03:03.000000000 +0700
+++ lz4-0.7.0.new/setup.py	2015-06-27 23:42:59.890180285 +0700
@@ -4,10 +4,11 @@
 from setuptools import setup, find_packages, Extension
 
 VERSION = (0, 7, 0)
+VERSION_STR = ".".join([str(x) for x in VERSION])
 
 setup(
     name='lz4',
-    version=".".join([str(x) for x in VERSION]),
+    version=VERSION_STR,
     description="LZ4 Bindings for Python",
     long_description=open('README.rst', 'r').read(),
     author='Steeve Morin',
@@ -26,6 +27,7 @@
             "-Wall",
             "-W",
             "-Wundef",
+            "-DVERSION=\"%s\"" % VERSION_STR,
             "-DLZ4_VERSION=\"r119\"",
         ])
     ],
diff -urN lz4-0.7.0/src/python-lz4.c lz4-0.7.0.new/src/python-lz4.c
--- lz4-0.7.0/src/python-lz4.c	2014-07-07 20:03:03.000000000 +0700
+++ lz4-0.7.0.new/src/python-lz4.c	2015-06-27 23:43:29.289958917 +0700
@@ -202,6 +202,9 @@
         INITERROR;
     }
 
+    PyModule_AddStringConstant(module, "VERSION", VERSION);
+    PyModule_AddStringConstant(module, "LZ4_VERSION", LZ4_VERSION);
+
 #if PY_MAJOR_VERSION >= 3
     return module;
 #endif
