--- a/lz4/__init__.py	2018-08-22 05:32:48.000000000 +0700
+++ b/lz4/__init__.py	2018-08-22 05:33:44.000000000 +0700
@@ -2,8 +2,9 @@
 try:
     __version__ = get_distribution(__name__).version
 except DistributionNotFound:
-    # package is not installed
-    pass
+    #when packaging things,
+    #get_distribution fails, and we just don't care:
+    __version__ = "2.1.0"
 
 VERSION = __version__
 
 