--- a/setup.py	2016-01-25 20:30:52.000000000 +0000+++ b/setup.py	2016-01-25 20:30:52.000000000 +0000@@ -27,7 +27,7 @@     extra_objects.append(os.path.join(LZO_DIR, "lzo.lib")) else:     libraries = ["lzo2"]-    include_dirs.append(os.environ.get("PREFIX", "/usr")+"/include/lzo")+    include_dirs.append("/usr/include/lzo")     ##library_dirs.append("/usr/local/lib")     ##runtime_library_dirs.append("/usr/local/lib") --- a/setup.py	2016-01-25 20:30:52.000000000 +0000
+++ b/setup.py	2016-01-25 20:30:52.000000000 +0000
@@ -27,7 +27,7 @@
     extra_objects.append(os.path.join(LZO_DIR, "lzo.lib"))
 else:
     libraries = ["lzo2"]
-    include_dirs.append(os.environ.get("PREFIX", "/usr")+"/include/lzo")
+    include_dirs.append("/usr/include/lzo")
     ##library_dirs.append("/usr/local/lib")
     ##runtime_library_dirs.append("/usr/local/lib")
 
