--- a/configure	2017-05-22 01:49:37.000000000 +0700
+++ b/configure	2017-06-12 02:37:01.000000000 +0700
@@ -20927,7 +20927,7 @@
 ac_config_headers="$ac_config_headers libtiff/tif_config.h libtiff/tiffconf.h"
 
 
-ac_config_files="$ac_config_files Makefile build/Makefile contrib/Makefile contrib/addtiffo/Makefile contrib/dbs/Makefile contrib/dbs/xtiff/Makefile contrib/iptcutil/Makefile contrib/mfs/Makefile contrib/pds/Makefile contrib/ras/Makefile contrib/stream/Makefile contrib/tags/Makefile contrib/win_dib/Makefile html/Makefile html/images/Makefile html/man/Makefile libtiff-4.pc libtiff/Makefile man/Makefile port/Makefile test/Makefile tools/Makefile"
+ac_config_files="$ac_config_files Makefile build/Makefile contrib/Makefile contrib/addtiffo/Makefile contrib/dbs/Makefile contrib/dbs/xtiff/Makefile contrib/iptcutil/Makefile contrib/mfs/Makefile contrib/pds/Makefile contrib/ras/Makefile contrib/stream/Makefile contrib/tags/Makefile contrib/win_dib/Makefile libtiff-4.pc libtiff/Makefile man/Makefile port/Makefile test/Makefile tools/Makefile"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -22095,9 +22095,6 @@
     "contrib/stream/Makefile") CONFIG_FILES="$CONFIG_FILES contrib/stream/Makefile" ;;
     "contrib/tags/Makefile") CONFIG_FILES="$CONFIG_FILES contrib/tags/Makefile" ;;
     "contrib/win_dib/Makefile") CONFIG_FILES="$CONFIG_FILES contrib/win_dib/Makefile" ;;
-    "html/Makefile") CONFIG_FILES="$CONFIG_FILES html/Makefile" ;;
-    "html/images/Makefile") CONFIG_FILES="$CONFIG_FILES html/images/Makefile" ;;
-    "html/man/Makefile") CONFIG_FILES="$CONFIG_FILES html/man/Makefile" ;;
     "libtiff-4.pc") CONFIG_FILES="$CONFIG_FILES libtiff-4.pc" ;;
     "libtiff/Makefile") CONFIG_FILES="$CONFIG_FILES libtiff/Makefile" ;;
     "man/Makefile") CONFIG_FILES="$CONFIG_FILES man/Makefile" ;;
--- a/Makefile.in	2017-05-22 01:49:35.000000000 +0700
+++ b/Makefile.in	2017-06-12 02:47:22.000000000 +0700
@@ -436,7 +436,7 @@
 	nmake.opt
 
 dist_doc_DATA = $(docfiles)
-SUBDIRS = port libtiff tools build contrib test man html
+SUBDIRS = port libtiff tools build contrib test man
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libtiff-4.pc
 
--- a/Makefile.am	2015-09-07 02:30:46.000000000 +0700
+++ b/Makefile.am	2017-06-12 02:46:47.000000000 +0700
@@ -61,7 +61,7 @@
 	rm -rf $(distdir)/_build/cmake
 	rm -rf $(distdir)/_inst/cmake
 
-SUBDIRS = port libtiff tools build contrib test man html
+SUBDIRS = port libtiff tools build contrib test man
 
 release:
 	(rm -f $(top_srcdir)/RELEASE-DATE && echo $(LIBTIFF_RELEASE_DATE) > $(top_srcdir)/RELEASE-DATE)

