diff --git a/cups/xpraforwarder b/cups/xpraforwarder
index 57d65582b..6fcbfc55a 100755
--- a/cups/xpraforwarder
+++ b/cups/xpraforwarder
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # xpraforwarder - A CUPS backend written in Python.
diff --git a/scripts/auth_dialog b/scripts/auth_dialog
index 831663c33..7929d0750 100755
--- a/scripts/auth_dialog
+++ b/scripts/auth_dialog
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import os
 import sys
diff --git a/scripts/xdg-open b/scripts/xdg-open
index d84b507d8..e1450d37c 100755
--- a/scripts/xdg-open
+++ b/scripts/xdg-open
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # This file is part of Xpra.
 # Copyright (C) 2017-2020 Antoine Martin <antoine@xpra.org>
 # Xpra is released under the terms of the GNU GPL v2, or, at your option, any
diff --git a/scripts/xpra b/scripts/xpra
index 36796ced7..008d0b260 100755
--- a/scripts/xpra
+++ b/scripts/xpra
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import os
 import sys
diff --git a/scripts/xpra_launcher b/scripts/xpra_launcher
index 0cbac5ed5..5743ea798 100755
--- a/scripts/xpra_launcher
+++ b/scripts/xpra_launcher
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import os
 import sys
diff --git a/scripts/xpra_signal_listener b/scripts/xpra_signal_listener
index b9cf743c2..b67c2eb12 100755
--- a/scripts/xpra_signal_listener
+++ b/scripts/xpra_signal_listener
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import os
 import sys
