Index: scripts/gnome-open
===================================================================
--- a/scripts/gnome-open	(revision 24151)
+++ b/scripts/gnome-open	(working copy)
@@ -1,4 +1,4 @@
-#!/usr/bin/sh
+#!/bin/sh
 # This file is part of Xpra.
 # Copyright (C) 2017 Antoine Martin <antoine@xpra.org>
 # Xpra is released under the terms of the GNU GPL v2, or, at your option, any
Index: scripts/gvfs-open
===================================================================
--- a/scripts/gvfs-open	(revision 24151)
+++ b/scripts/gvfs-open	(working copy)
@@ -1,4 +1,4 @@
-#!/usr/bin/sh
+#!/bin/sh
 # This file is part of Xpra.
 # Copyright (C) 2017 Antoine Martin <antoine@xpra.org>
 # Xpra is released under the terms of the GNU GPL v2, or, at your option, any
Index: scripts/xpra_udev_product_version
===================================================================
--- a/scripts/xpra_udev_product_version	(revision 24151)
+++ b/scripts/xpra_udev_product_version	(working copy)
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/bin/bash
 # This file is part of Xpra.
 # Copyright (C) 2017 Antoine Martin <antoine@xpra.org>
 # Xpra is released under the terms of the GNU GPL v2, or, at your option, any
