Index: src/xpra/codecs/dec_avcodec2/decoder.pyx
===================================================================
--- a/xpra/codecs/dec_avcodec2/decoder.pyx	(revision 8396)
+++ b/xpra/codecs/dec_avcodec2/decoder.pyx	(working copy)
@@ -58,8 +58,8 @@
     int AV_PIX_FMT_YUV422P
     int AV_PIX_FMT_YUV444P
     int AV_PIX_FMT_RGB24
-    int AV_PIX_FMT_0RGB
-    int AV_PIX_FMT_BGR0
+    #int AV_PIX_FMT_0RGB
+    #int AV_PIX_FMT_BGR0
     int AV_PIX_FMT_ARGB
     int AV_PIX_FMT_BGRA
     int AV_PIX_FMT_GBRP
@@ -92,7 +92,7 @@
 
     AVPixelFormat PIX_FMT_NONE
     AVCodecID AV_CODEC_ID_H264
-    AVCodecID AV_CODEC_ID_H265
+    #AVCodecID AV_CODEC_ID_H265
     AVCodecID AV_CODEC_ID_VP8
     AVCodecID AV_CODEC_ID_VP9
 
@@ -119,8 +119,8 @@
             "YUV422P"   : AV_PIX_FMT_YUV422P,
             "YUV444P"   : AV_PIX_FMT_YUV444P,
             "RGB"       : AV_PIX_FMT_RGB24,
-            "XRGB"      : AV_PIX_FMT_0RGB,
-            "BGRX"      : AV_PIX_FMT_BGR0,
+            #"XRGB"      : AV_PIX_FMT_0RGB,
+            #"BGRX"      : AV_PIX_FMT_BGR0,
             "ARGB"      : AV_PIX_FMT_ARGB,
             "BGRA"      : AV_PIX_FMT_BGRA,
             "GBRP"      : AV_PIX_FMT_GBRP,
@@ -139,8 +139,8 @@
     CODECS.append("vp8")
 if avcodec_find_decoder(AV_CODEC_ID_VP9)!=NULL:
     CODECS.append("vp9")
-if avcodec_find_decoder(AV_CODEC_ID_H265)!=NULL:
-    CODECS.append("h265")
+#if avcodec_find_decoder(AV_CODEC_ID_H265)!=NULL:
+#    CODECS.append("h265")
 log("avcodec2.init_module: CODECS=%s", CODECS)
 
 
@@ -313,8 +313,8 @@
         cdef AVCodecID CodecID
         if self.encoding=="h264":
             CodecID = AV_CODEC_ID_H264
-        elif self.encoding=="h265":
-            CodecID = AV_CODEC_ID_H265
+        #elif self.encoding=="h265":
+        #    CodecID = AV_CODEC_ID_H265
         elif self.encoding=="vp8":
             CodecID = AV_CODEC_ID_VP8
         elif self.encoding=="vp9":
