Index: xpra/codecs/csc_swscale/colorspace_converter.pyx
===================================================================
--- a/xpra/codecs/csc_swscale/colorspace_converter.pyx	(revision 8996)
+++ b/xpra/codecs/csc_swscale/colorspace_converter.pyx	(working copy)
@@ -78,14 +78,14 @@
 COLORSPACES = []
 #keeping this array in scope ensures the strings don't go away!
 FORMAT_OPTIONS = [
-    ("AV_PIX_FMT_RGB24",    (3, 0, 0, 0),       (1, 0, 0, 0),       "RGB"  ),
-    ("AV_PIX_FMT_BGR24",    (3, 0, 0, 0),       (1, 0, 0, 0),       "BGR"  ),
-    ("AV_PIX_FMT_ARGB",     (4, 0, 0, 0),       (1, 0, 0, 0),       "XRGB"  ),
-    ("AV_PIX_FMT_BGRA",     (4, 0, 0, 0),       (1, 0, 0, 0),       "BGRX"  ),
-    ("AV_PIX_FMT_YUV420P",  (1, 0.5, 0.5, 0),   (1, 0.5, 0.5, 0),   "YUV420P"),
-    ("AV_PIX_FMT_YUV422P",  (1, 0.5, 0.5, 0),   (1, 1, 1, 0),       "YUV422P"),
-    ("AV_PIX_FMT_YUV444P",  (1, 1, 1, 0),       (1, 1, 1, 0),       "YUV444P"),
-    ("AV_PIX_FMT_GBRP",     (1, 1, 1, 0),       (1, 1, 1, 0),       "GBRP"   )
+    ("PIX_FMT_RGB24",    (3, 0, 0, 0),       (1, 0, 0, 0),       "RGB"  ),
+    ("PIX_FMT_BGR24",    (3, 0, 0, 0),       (1, 0, 0, 0),       "BGR"  ),
+    ("PIX_FMT_ARGB",     (4, 0, 0, 0),       (1, 0, 0, 0),       "XRGB"  ),
+    ("PIX_FMT_BGRA",     (4, 0, 0, 0),       (1, 0, 0, 0),       "BGRX"  ),
+    ("PIX_FMT_YUV420P",  (1, 0.5, 0.5, 0),   (1, 0.5, 0.5, 0),   "YUV420P"),
+    ("PIX_FMT_YUV422P",  (1, 0.5, 0.5, 0),   (1, 1, 1, 0),       "YUV422P"),
+    ("PIX_FMT_YUV444P",  (1, 1, 1, 0),       (1, 1, 1, 0),       "YUV444P"),
+    ("PIX_FMT_GBRP",     (1, 1, 1, 0),       (1, 1, 1, 0),       "GBRP"   )
      ]
 FORMATS = {}
 for av_enum_name, width_mult, height_mult, pix_fmt in FORMAT_OPTIONS:
Index: xpra/codecs/csc_swscale/constants.txt
===================================================================
--- a/xpra/codecs/csc_swscale/constants.txt	(revision 8996)
+++ b/xpra/codecs/csc_swscale/constants.txt	(working copy)
@@ -10,15 +10,15 @@
 ## dictionary through which the constants can be accessed like
 ## constants["AV_..."].
 
-AV_PIX_FMT_YUV420P
-AV_PIX_FMT_YUV422P
-AV_PIX_FMT_YUV444P
-AV_PIX_FMT_RGB24
-AV_PIX_FMT_ARGB
-AV_PIX_FMT_BGRA
-AV_PIX_FMT_GBRP
-AV_PIX_FMT_RGB24
-AV_PIX_FMT_BGR24
+PIX_FMT_YUV420P
+PIX_FMT_YUV422P
+PIX_FMT_YUV444P
+PIX_FMT_RGB24
+PIX_FMT_ARGB
+PIX_FMT_BGRA
+PIX_FMT_GBRP
+PIX_FMT_RGB24
+PIX_FMT_BGR24
 
 SWS_ACCURATE_RND
 SWS_BICUBIC
