Index: src/xpra/codecs/dec_avcodec/decoder.pyx
===================================================================
--- a/xpra/codecs/dec_avcodec/decoder.pyx	(revision 4745)
+++ b/xpra/codecs/dec_avcodec/decoder.pyx	(working copy)
@@ -388,8 +388,8 @@
         self.codec_ctx.get_buffer = avcodec_get_buffer
         self.codec_ctx.release_buffer = avcodec_release_buffer
         self.codec_ctx.thread_safe_callbacks = 1
-        self.codec_ctx.thread_type = 2      #FF_THREAD_SLICE: allow more than one thread per frame
-        self.codec_ctx.thread_count = 0     #auto
+        self.codec_ctx.thread_type = 1      #FF_THREAD_SLICE: allow more than one thread per frame
+        self.codec_ctx.thread_count = 1
         self.codec_ctx.flags2 |= CODEC_FLAG2_FAST   #may cause "no deblock across slices" - which should be fine
         r = avcodec_open2(self.codec_ctx, self.codec, NULL)
         if r<0:
