Index: xpra/codecs/dec_avcodec/decoder.pyx
===================================================================
--- a/xpra/codecs/dec_avcodec/decoder.pyx	(revision 4990)
+++ b/xpra/codecs/dec_avcodec/decoder.pyx	(working copy)
@@ -114,7 +114,7 @@
 def get_type():
     return "avcodec"
 
-MIN_AVCODEC_VERSION = 54
+MIN_AVCODEC_VERSION = 53
 COLORSPACES = None
 FORMAT_TO_ENUM = {}
 ENUM_TO_FORMAT = {}
