Index: fs/etc/pam.d/xpra
===================================================================
--- a/fs/etc/pam.d/xpra	(revision 18636)
+++ b/fs/etc/pam.d/xpra	(working copy)
@@ -12,6 +12,5 @@
 session    required     pam_selinux.so open
 session    required     pam_namespace.so
 session    optional     pam_keyinit.so force revoke
-session    include      system-auth
-session    include      postlogin
+session    include      common-session
 -session   optional     pam_ck_connector.so
