Index: ./packaging/debian/xpra/xpra-server.service
===================================================================
--- a/packaging/debian/xpra/xpra-server.service	(revision 24973)
+++ b/packaging/debian/xpra/xpra-server.service	(working copy)
@@ -7,7 +7,7 @@
 
 [Service]
 Type=simple
-EnvironmentFile=-/etc/sysconfig/xpra
+EnvironmentFile=-/etc/default/xpra
 ExecStart=/usr/bin/xpra proxy :14500 --daemon=no \
     --tcp-auth=${TCP_AUTH} \
     --ssl-cert=/etc/xpra/ssl-cert.pem --ssl=on \
