--- a/CMakeLists.txt	2023-02-03 11:54:04.943595524 +0700
+++ b/CMakeLists.txt	2023-02-03 11:55:13.546107333 +0700
@@ -35,10 +35,6 @@
 
 SET_TARGET_PROPERTIES	( ${ly_lib_shared} PROPERTIES VERSION "0.0.0" SOVERSION 0 )
 
-# this creates the conversion tool
-ADD_EXECUTABLE			( yuvconvert ${ly_base_dir}/util/yuvconvert.cc )
-TARGET_LINK_LIBRARIES	( yuvconvert ${ly_lib_shared} )
-
 # this creates the yuvconstants tool
 ADD_EXECUTABLE      ( yuvconstants ${ly_base_dir}/util/yuvconstants.c )
 TARGET_LINK_LIBRARIES  ( yuvconstants ${ly_lib_static} )
