Index: setup.py
===================================================================
--- a/setup.py	(revision 10205)
+++ b/setup.py	(working copy)
@@ -611,7 +611,7 @@
             elif get_gcc_version()>=[4, 4]:
                 eifd = ["-Werror",
                         #CentOS 6.x gives us some invalid warnings in nvenc, ignore those:
-                        #"-Wno-error=uninitialized",
+                        "-Wno-error=uninitialized",
                         #needed on Debian and Ubuntu to avoid this error:
                         #/usr/include/gtk-2.0/gtk/gtkitemfactory.h:47:1: error: function declaration isn't a prototype [-Werror=strict-prototypes]
                         #"-Wno-error=strict-prototypes",
