Index: service/xpra.service
===================================================================
--- a/service/xpra.service	(revision 16054)
+++ b/service/xpra.service	(working copy)
@@ -13,11 +13,11 @@
 SuccessExitStatus=0 143
 Restart=on-abnormal
 PIDFile=/run/xpra.pid
-ProtectSystem=strict
-ReadWritePaths=/run /tmp
+#ProtectSystem=strict
+#ReadWritePaths=/run /tmp
 #PrivateDevices=true
-ProtectKernelTunables=true
-ProtectControlGroups=true
+#ProtectKernelTunables=true
+#ProtectControlGroups=true
 
 [Install]
 WantedBy=multi-user.target
