Index: selinux/cups_xpra/cups_xpra.te
===================================================================
--- selinux/cups_xpra/cups_xpra.te	(revision 13327)
+++ selinux/cups_xpra/cups_xpra.te	(working copy)
@@ -11,7 +11,7 @@
 	type udev_var_run_t;
 	role system_r;
 	class capability dac_override;
-	class process { signal transition sigchld };
+	class process { signal transition sigchld execmem };
 	class fd { use };
 	class dir { getattr search open read lock ioctl };
 	class fifo_file { getattr read write append ioctl lock };
@@ -42,6 +42,8 @@
 #logging_send_audit_msgs(cupsd_t)
 logging_send_syslog_msg(cups_xpra_t)
 
+allow cups_xpra_t self:process execmem;
+
 allow cups_xpra_t self:capability dac_override;
 allow cups_xpra_t self:netlink_kobject_uevent_socket create_socket_perms;
 
