Index: xpra/server/window/motion.pyx
===================================================================
--- a/xpra/server/window/motion.pyx	(revision 13756)
+++ b/xpra/server/window/motion.pyx	(working copy)
@@ -67,9 +67,9 @@
     return crcs
 
 
-DEF MAXINT64 = 2**63
-DEF MAXUINT64 = 2**64
-DEF MASK64 = 2**64-1
+MAXINT64 = 2**63
+MAXUINT64 = 2**64
+MASK64 = 2**64-1
 cdef inline castint64(v):
     if v>=MAXINT64:
         return v-MAXUINT64
