Index: xpra/codecs/enc_x264/encoder.pyx
===================================================================
--- a/xpra/codecs/enc_x264/encoder.pyx	(revision 4222)
+++ b/xpra/codecs/enc_x264/encoder.pyx	(working copy)
@@ -69,6 +69,9 @@
         int b_open_gop
         int b_bluray_compat
 
+        #not all builds support this:
+        int b_opencl            #use OpenCL when available
+
     ctypedef struct x264_t:
         pass
     ctypedef struct x264_nal_t:
@@ -236,6 +239,7 @@
         param.i_keyint_min = 999999
         param.b_intra_refresh = 0   #no intra refresh
         param.b_open_gop = 1        #allow open gop
+        param.b_opencl = 1          #use opencl
         x264_param_apply_profile(&param, self.profile)
         self.context = x264_encoder_open(&param)
         assert self.context!=NULL,  "context initialization failed for format %s" % self.src_format
