Index: xpra/sound/gstreamer_util.py
===================================================================
--- a/xpra/sound/gstreamer_util.py	(revision 13756)
+++ b/xpra/sound/gstreamer_util.py	(working copy)
@@ -45,7 +45,7 @@
 OSX = sys.platform.startswith("darwin")
 
 ALLOW_SOUND_LOOP = envbool("XPRA_ALLOW_SOUND_LOOP", False)
-GSTREAMER1 = envbool("XPRA_GSTREAMER1", True)
+GSTREAMER1 = envbool("XPRA_GSTREAMER1", False)
 PULSEAUDIO_DEVICE_NAME = os.environ.get("XPRA_PULSEAUDIO_DEVICE_NAME", "")
 USE_DEFAULT_DEVICE = envbool("XPRA_USE_DEFAULT_DEVICE", True)
 def force_enabled(codec_name):
