Index: xpra/codecs/dec_avcodec2/decoder.pyx
===================================================================
--- xpra/codecs/dec_avcodec2/decoder.pyx	(revision 6508)
+++ xpra/codecs/dec_avcodec2/decoder.pyx	(working copy)
@@ -57,8 +57,8 @@
     int AV_PIX_FMT_YUV422P
     int AV_PIX_FMT_YUV444P
     int AV_PIX_FMT_RGB24
-    int AV_PIX_FMT_0RGB
-    int AV_PIX_FMT_BGR0
+    #int AV_PIX_FMT_0RGB
+    #int AV_PIX_FMT_BGR0
     int AV_PIX_FMT_ARGB
     int AV_PIX_FMT_BGRA
     int AV_PIX_FMT_GBRP
@@ -94,7 +94,7 @@
 
     AVPixelFormat PIX_FMT_NONE
     AVCodecID AV_CODEC_ID_H264
-    AVCodecID AV_CODEC_ID_H265
+    #AVCodecID AV_CODEC_ID_H265
     AVCodecID AV_CODEC_ID_VP8
     AVCodecID AV_CODEC_ID_VP9
 
@@ -121,8 +121,8 @@
             "YUV422P"   : AV_PIX_FMT_YUV422P,
             "YUV444P"   : AV_PIX_FMT_YUV444P,
             "RGB"       : AV_PIX_FMT_RGB24,
-            "XRGB"      : AV_PIX_FMT_0RGB,
-            "BGRX"      : AV_PIX_FMT_BGR0,
+            #"XRGB"      : AV_PIX_FMT_0RGB,
+            #"BGRX"      : AV_PIX_FMT_BGR0,
             "ARGB"      : AV_PIX_FMT_ARGB,
             "BGRA"      : AV_PIX_FMT_BGRA,
             "GBRP"      : AV_PIX_FMT_GBRP,
@@ -141,8 +141,8 @@
     CODECS.append("vp8")
 if avcodec_find_decoder(AV_CODEC_ID_VP9)!=NULL:
     CODECS.append("vp9")
-if avcodec_find_decoder(AV_CODEC_ID_H265)!=NULL:
-    CODECS.append("h265")
+#if avcodec_find_decoder(AV_CODEC_ID_H265)!=NULL:
+#    CODECS.append("h265")
 log("avcodec2.init_module: CODECS=%s", CODECS)
 
 
@@ -316,7 +316,7 @@
                 log.error("codec H264 not found!")
                 return  False
         elif self.encoding=="h265":
-            self.codec = avcodec_find_decoder(AV_CODEC_ID_H265)
+            self.codec = NULL       #avcodec_find_decoder(AV_CODEC_ID_H265)
             if self.codec==NULL:
                 log.error("codec H265 (HEVC) not found!")
                 return  False
