Index: xpra/codecs/dec_avcodec/decoder.pyx
===================================================================
--- a/xpra/codecs/dec_avcodec/decoder.pyx	(revision 4080)
+++ b/xpra/codecs/dec_avcodec/decoder.pyx	(working copy)
@@ -76,8 +76,8 @@
     AVCodec *avcodec_find_decoder(AVCodecID id)
     AVCodecContext *avcodec_alloc_context3(const AVCodec *codec)
     int avcodec_open2(AVCodecContext *avctx, const AVCodec *codec, AVDictionary **options)
-    AVFrame *avcodec_alloc_frame()
-    void avcodec_free_frame(AVFrame **frame)
+    #AVFrame *avcodec_alloc_frame()
+    #void avcodec_free_frame(AVFrame **frame)
     int avcodec_close(AVCodecContext *avctx)
     void av_free(void *ptr)
 
@@ -340,7 +340,7 @@
             error("could not open codec")
             self.clean_decoder()
             return  False
-        self.frame = avcodec_alloc_frame()
+        self.frame = <AVFrame *> xmemalign(sizeof(AVFrame))
         if self.frame==NULL:
             error("could not allocate an AVFrame for decoding")
             self.clean_decoder()
@@ -375,8 +375,8 @@
 
         debug("clean_decoder() freeing AVFrame: %s", hex(<unsigned long> self.frame))
         if self.frame!=NULL:
-            avcodec_free_frame(&self.frame)
-            #redundant: self.frame = NULL
+            #free(&self.frame)   #freeing here causes a double free on Ubuntu, avcodec_close probably does it too
+            self.frame = NULL
 
         cdef unsigned long ctx_key          #@DuplicatedSignature
         debug("clean_decoder() freeing AVCodecContext: %s", hex(<unsigned long> self.codec_ctx))
