Index: xpra/server/window_source.py
===================================================================
--- a/xpra/server/window_source.py	(revision 4677)
+++ b/xpra/server/window_source.py	(working copy)
@@ -20,7 +20,7 @@
 
 DELTA = os.environ.get("XPRA_DELTA", "1")=="1"
 MAX_DELTA_SIZE = int(os.environ.get("XPRA_MAX_DELTA_SIZE", "10000"))
-PIL_CAN_OPTIMIZE = os.environ.get("XPRA_PIL_OPTIMIZE", "1")=="1"
+PIL_CAN_OPTIMIZE = os.environ.get("XPRA_PIL_OPTIMIZE", "0")=="1"
 ALLOW_ALPHA = os.environ.get("XPRA_ALLOW_ALPHA", "1")=="1"
 
 XPRA_DAMAGE_DEBUG = os.environ.get("XPRA_DAMAGE_DEBUG", "0")!="0"
