Index: selinux/xpra_socketactivation/xpra_socketactivation.te
===================================================================
--- a/selinux/xpra_socketactivation/xpra_socketactivation.te	(revision 19805)
+++ b/selinux/xpra_socketactivation/xpra_socketactivation.te	(working copy)
@@ -104,7 +104,7 @@
 
 #probe Xorg binary suid status (centos only?):
 allow xpra_t xserver_exec_t:file { getattr execute };
-allow xpra_t ldconfig_exec_t:file { execute execute_no_trans getattr open read map };
+allow xpra_t ldconfig_exec_t:file { execute execute_no_trans getattr open read };
 
 #dbus / mdns:
 allow xpra_t avahi_t:dbus send_msg;
@@ -156,5 +156,5 @@
 allow xpra_t sysfs_t:dir read;
 allow xpra_t user_tmp_t:sock_file write;
 #proxy instance mmap:
-allow xpra_t tmpfs_t:file map;
-allow xpra_t bin_t:file map;
+#allow xpra_t tmpfs_t:file map;
+#allow xpra_t bin_t:file map;
