Index: setup.py
===================================================================
--- a/setup.py	(revision 3160)
+++ b/setup.py	(working copy)
@@ -809,10 +809,13 @@
 
 
 if vpx_ENABLED:
+    vpxstatic = {'include_dirs': ["/usr/local/include"],
+                'library_dirs': ["/usr/local/lib"],
+                'extra_link_args': ["-Wl,-soname,vpxlib.so", "-Wl,-Bstatic", "-Wl,-Bsymbolic", "-lvpx", "-lswscale", "-lavcodec", "-lavutil", "-Wl,-Bdynamic"] }
     packages.append("xpra.codecs.vpx")
     cython_add(Extension("xpra.codecs.vpx.codec",
                 ["xpra/codecs/vpx/codec.pyx", "xpra/codecs/vpx/vpxlib.c"],
-                **pkgconfig(["libvpx", "vpx"], "libswscale", "libavcodec")
+                **vpxstatic
                 ), min_version=(0, 16))
 elif sys.platform.startswith("win"):
     py2exe_excludes.append("xpra.codecs.vpx")
