diff -urN a/src/dummy_driver.c b/src/dummy_driver.c
--- a/src/dummy_driver.c	2022-04-06 13:43:22.348626915 +0700
+++ b/src/dummy_driver.c	2022-04-06 13:43:37.175533701 +0700
@@ -320,6 +320,7 @@
 	case 15:
 	case 16:
 	case 24:
+        case 30:
 	    break;
 	default:
 	    xf86DrvMsg(pScrn->scrnIndex, X_ERROR,
@@ -334,8 +335,8 @@
 	pScrn->rgbBits = 8;
 
     /* Get the depth24 pixmap format */
-    if (pScrn->depth == 24 && pix24bpp == 0)
-	pix24bpp = xf86GetBppFromDepth(pScrn, 24);
+    if (pScrn->depth >= 24 && pix24bpp == 0)
+	pix24bpp = xf86GetBppFromDepth(pScrn, pScrn->depth);
 
     /*
      * This must happen after pScrn->display has been set because
@@ -626,7 +627,7 @@
     if(!miCreateDefColormap(pScreen))
 	return FALSE;
 
-    if (!xf86HandleColormaps(pScreen, 256, pScrn->rgbBits,
+    if (!xf86HandleColormaps(pScreen, 1024, pScrn->rgbBits,
                          DUMMYLoadPalette, NULL, 
                          CMAP_PALETTED_TRUECOLOR 
 			     | CMAP_RELOAD_ON_MODE_SWITCH))
diff -urN xf86-video-dummy-0.3.8.bak/src/dummy.h xf86-video-dummy-0.3.8/src/dummy.h
--- xf86-video-dummy-0.3.8.bak/src/dummy.h	2022-04-06 13:43:20.004641651 +0700
+++ xf86-video-dummy-0.3.8/src/dummy.h	2022-04-06 13:43:36.995534832 +0700
@@ -69,7 +69,7 @@
     int overlay_offset;
     int videoKey;
     int interlace;
-    dummy_colors colors[256];
+    dummy_colors colors[1024];
     pointer* FBBase;
     Bool        (*CreateWindow)() ;     /* wrapped CreateWindow */
     Bool prop;
