
--- a/gio/gdummyfile.c	Sat Jun 28 10:02:43 2014
+++ b/gio/gdummyfile.c	Tue Dec  9 13:17:43 2014
@@ -441,7 +441,7 @@
 		 const gchar *escaped_string_End,
 		 const gchar *illegal_characters)
 {
-  const gchar *in;
+  const gchar *inchar;
   gchar *out, *result;
   gint character;
   
@@ -454,19 +454,19 @@
   result = g_malloc (escaped_string_end - escaped_string + 1);
 	
   out = result;
-  for (in = escaped_string; in < escaped_string_end; in++) 
+  for (inchar = escaped_string; inchar < escaped_string_end; inchar++) 
     {
-      character = *in;
-      if (*in == '%') 
+      character = *inchar;
+      if (*inchar == '%') 
         {
-          in++;
-          if (escaped_string_end - in < 2)
+          inchar++;
+          if (escaped_string_end - inchar < 2)
 	    {
 	      g_free (result);
 	      return NULL;
 	    }
       
-          character = unescape_character (in);
+          character = unescape_character (inchar);
       
           /* Check for an illegal character. We consider '\0' illegal here. */
           if (character <= 0 ||
@@ -476,7 +476,7 @@
 	      g_free (result);
 	      return NULL;
 	    }
-          in++; /* The other char will be eaten in the loop header */
+          inchar++; /* The other char will be eaten in the loop header */
         }
       *out++ = (char)character;
     }
@@ -516,7 +516,7 @@
 _g_decode_uri (const char *uri)
 {
   GDecodedUri *decoded;
-  const char *p, *in, *hier_part_start, *hier_part_end, *query_start, *fragment_start;
+  const char *p, *inchar, *hier_part_start, *hier_part_end, *query_start, *fragment_start;
   char *out;
   char c;
 
@@ -551,8 +551,8 @@
   
   decoded->scheme = g_malloc (p - uri);
   out = decoded->scheme;
-  for (in = uri; in < p - 1; in++)
-    *out++ = g_ascii_tolower (*in);
+  for (inchar = uri; inchar < p - 1; inchar++)
+    *out++ = g_ascii_tolower (*inchar);
   *out = 0;
 
   hier_part_start = p;

Diff finished.  Sun Dec 21 10:08:53 2014
