diff --git a/CMakeLists.txt b/CMakeLists.txt
index ed4948f0..8be3402a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -89,3 +89,6 @@ INSTALL ( DIRECTORY ${PROJECT_SOURCE_DIR}/include/		DESTINATION include )
 # create the .deb and .rpm packages using cpack
 INCLUDE ( CM_linux_packages.cmake )
 
+# generate pkgconfig file:
+install ( CODE "execute_process(COMMAND sh ${PROJECT_SOURCE_DIR}/gen_pc.sh)")
+
Index: gen_pc.sh
===================================================================
--- a/gen_pc.sh	(nonexistent)
+++ b/gen_pc.sh	(working copy)
@@ -0,0 +1,10 @@
+cat > ${JHBUILD_PREFIX}/lib/pkgconfig/libyuv.pc <<EOF
+prefix=${JHBUILD_PREFIX}
+libdir=${prefix}/lib
+includedir=${prefix}/include
+
+Name: libyuv
+Description: YUV conversion and scaling functionality library
+Version: 0
+Libs: -lyuv
+EOF
