From 44f04fd3046043ed31369025f34353c4e0e5c1cd Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Tue, 25 Sep 2012 08:54:36 -0400
Subject: [PATCH] Remove mibstore.h

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
 src/dummy_driver.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/dummy_driver.c b/src/dummy_driver.c
index 62066d6..6062c39 100644
--- a/src/dummy_driver.c
+++ b/src/dummy_driver.c
@@ -14,9 +14,6 @@
 /* All drivers initialising the SW cursor need this */
 #include "mipointer.h"
 
-/* All drivers implementing backing store need this */
-#include "mibstore.h"
-
 /* All drivers using the mi colormap manipulation need this */
 #include "micmap.h"
 
@@ -617,7 +614,6 @@ DUMMYScreenInit(SCREEN_INIT_ARGS_DECL)
 		   , lines - pScrn->virtualY);
     }
 
-    miInitializeBackingStore(pScreen);
     xf86SetBackingStore(pScreen);
     xf86SetSilkenMouse(pScreen);
 	
-- 
1.8.0.2
