diff --git a/src/dummy_driver.c b/src/dummy_driver.c
index 6062c39..470e125 100644
--- a/src/dummy_driver.c
+++ b/src/dummy_driver.c
@@ -281,7 +281,7 @@ DUMMYPreInit(ScrnInfoPtr pScrn, int flags)
     ClockRangePtr clockRanges;
     int i;
     DUMMYPtr dPtr;
-    int maxClock = 230000;
+    int maxClock = 300000;
     GDevPtr device = xf86GetEntityInfo(pScrn->entityList[0])->device;
 
     if (flags & PROBE_DETECT) 
@@ -391,7 +391,7 @@ DUMMYPreInit(ScrnInfoPtr pScrn, int flags)
     clockRanges->next = NULL;
     clockRanges->ClockMulFactor = 1;
     clockRanges->minClock = 11000;   /* guessed §§§ */
-    clockRanges->maxClock = 300000;
+    clockRanges->maxClock = maxClock;
     clockRanges->clockIndex = -1;		/* programmable */
     clockRanges->interlaceAllowed = TRUE; 
     clockRanges->doubleScanAllowed = TRUE;
