--- a/src/dummy_driver.c	2016-10-26 15:31:33.745940560 +0700
+++ b/src/dummy_driver.c	2016-10-26 15:31:05.313666076 +0700
@@ -842,8 +842,9 @@
         if (! ValidAtom(VFB_PROP))
             VFB_PROP = MakeAtom(VFB_PROP_NAME, strlen(VFB_PROP_NAME), 1);
 
-        ret = ChangeWindowProperty(pWinRoot, VFB_PROP, XA_STRING, 
-		8, PropModeReplace, (int)4, (pointer)"TRUE", FALSE);
+	ret = dixChangeWindowProperty(serverClient, pWinRoot, VFB_PROP,
+                                      XA_STRING, 8, PropModeReplace,
+                                      (int)4, (pointer)"TRUE", FALSE);
 	if( ret != Success)
 		ErrorF("Could not set VFB root window property");
         dPtr->prop = TRUE;
