--- a/Cython/Compiler/FlowControl.py	2015-08-09 00:54:59.000000000 +0700
+++ b/Cython/Compiler/FlowControl.py	2015-08-19 09:26:02.836996013 +0700
@@ -2,13 +2,10 @@
 
 import cython
 cython.declare(PyrexTypes=object, ExprNodes=object, Nodes=object,
-               Builtin=object, InternalError=object,
-               error=object, warning=object, deepcopy=object,
+               Builtin=object, InternalError=object, error=object, warning=object,
                py_object_type=object, unspecified_type=object,
                object_expr=object, fake_rhs_expr=object, TypedExprNode=object)
 
-from copy import deepcopy
-
 from . import Builtin
 from . import ExprNodes
 from . import Nodes
@@ -328,17 +325,6 @@
         self.is_deletion = False
         self.inferred_type = None
 
-    def __deepcopy__(self, memo):
-        ass = NameAssignment.__new__(type(self))
-        ass.lhs = deepcopy(self.lhs, memo)
-        ass.rhs = deepcopy(self.rhs, memo)
-        ass.entry = self.entry
-        ass.refs = deepcopy(self.refs, memo)
-        ass.is_arg = self.is_arg
-        ass.is_deletion = self.is_deletion
-        ass.inferred_type = self.inferred_type
-        return ass
-
     def __repr__(self):
         return '%s(entry=%r)' % (self.__class__.__name__, self.entry)
 
@@ -413,12 +399,6 @@
     def __repr__(self):
         return '%s(entry=%r)' % (self.__class__.__name__, self.entry)
 
-    def __deepcopy__(self, memo):
-        ref = NameReference.__new__(type(self))
-        ref.node = deepcopy(self.node, memo)
-        ref.entry = self.entry
-        ref.pos = self.node.pos
-
 
 class ControlFlowState(list):
     # Keeps track of Node's entry assignments
