--- a/setup.py	2017-03-13 13:35:21.531015209 +0700
+++ b/setup.py	2017-03-13 13:35:46.792826355 +0700
@@ -144,8 +144,8 @@
         lz4block,
         lz4frame
     ],
-    tests_require=["nose>=1.0"],
-    test_suite = "nose.collector",
+    #tests_require=["nose>=1.0"],
+    #test_suite = "nose.collector",
     classifiers=[
         'Development Status :: 5 - Production/Stable',
         'License :: OSI Approved :: BSD License',
