--- a/setup.py	2016-10-09 19:25:33.964304313 +0700
+++ b/setup.py	2016-10-09 19:25:44.395144309 +0700
@@ -40,8 +40,8 @@
         extra_compile_args=extra_compile_args,
         define_macros=define_macros)
     ],
-    setup_requires=["nose>=1.3.0"],
-    test_suite='nose.collector',
+    #setup_requires=["nose>=1.3.0"],
+    #test_suite='nose.collector',
     classifiers=[
         'Development Status :: 4 - Beta',
         'License :: OSI Approved :: BSD License',
