--- a/src/cpp/cuda.hpp	2019-01-13 23:51:24.718875847 +0700
+++ b/src/cpp/cuda.hpp	2019-01-13 23:52:02.315630622 +0700
@@ -497,7 +497,7 @@
   inline
   void init(unsigned int flags)
   {
-    CUDAPP_CALL_GUARDED(cuInit, (flags));
+    CUDAPP_CALL_GUARDED_THREADED(cuInit, (flags));
   }
 
   inline
@@ -822,7 +822,7 @@
     context::prepare_context_switch();
 
     CUcontext ctx;
-    CUDAPP_CALL_GUARDED(cuCtxCreate, (&ctx, flags, m_device));
+    CUDAPP_CALL_GUARDED_THREADED(cuCtxCreate, (&ctx, flags, m_device));
     boost::shared_ptr<context> result(new context(ctx));
     context_stack::get().push(result);
     return result;
