--- xpra/xposix/__init__.py	2010-11-20 23:18:07.153231001 +0700
+++ xpra/xposix/__init__.py	2011-03-26 04:06:29.940353667 +0700
@@ -6,7 +6,7 @@
 # Platform-specific code for Posix systems with X11 display.
 
 import sys
-XPRA_LOCAL_SERVERS_SUPPORTED = True
+XPRA_LOCAL_SERVERS_SUPPORTED = False
 DEFAULT_SSH_CMD = "ssh"
 GOT_PASSWORD_PROMPT_SUGGESTION = "Perhaps you need to set up your ssh agent?\n"
