Index: setup.py
===================================================================
--- setup.py	(revision 642)
+++ setup.py	(working copy)
@@ -64,14 +64,14 @@
                 ["xpra/wait_for_x_server.pyx"],
                 **pkgconfig("x11")
                 ),
-      Extension("xpra.x264.encoder",
-                ["xpra/x264/encoder.pyx", "xpra/x264/x264lib.c"],
-                **pkgconfig("x264", "libswscale", "libavcodec")
-                ),
-      Extension("xpra.x264.decoder",
-                ["xpra/x264/decoder.pyx", "xpra/x264/x264lib.c"],
-                **pkgconfig("x264", "libswscale", "libavcodec")
-                ),
+      #Extension("xpra.x264.encoder",
+      #          ["xpra/x264/encoder.pyx", "xpra/x264/x264lib.c"],
+      #          **pkgconfig("x264", "libswscale", "libavcodec")
+      #          ),
+      #Extension("xpra.x264.decoder",
+      #          ["xpra/x264/decoder.pyx", "xpra/x264/x264lib.c"],
+      #          **pkgconfig("x264", "libswscale", "libavcodec")
+      #          ),
       ]
 
     cmdclass = {'build_ext': build_ext}
